### PR TITLE
Only create a new observable query if options have changed

### PR DESCRIPTION
**Pull request by HofmannZ**
### Breaking changes

n/a

#### Fixes / Enhancements

- Only create a new observable query if options have changed. @HofmannZ

#### Docs

n/a

### CODE REVIEW

File: lib/src/core/query_options.dart [hunk 1]
```diff
@@ -136,4 +136,56 @@ class WatchQueryOptions extends BaseOptions {
...
+    if (a.pollInterval != b.pollInterval) {
+      return true;
+    }
+
+    if (a.fetchResults != b.fetchResults) {
+      return true;
+    }
+
+    // compare variables last, because maps take more time
+    return _areDifferentVariables(a.variables, b.variables);
```
Comment (eusdima) [inline lib/src/core/query_options.dart]: Nice optimization

File: lib/src/core/query_options.dart [hunk 2]
```diff
@@ -136,4 +136,56 @@ class WatchQueryOptions extends BaseOptions {
...
+    Map b,
+  ) {
+    if (a.length != b.length) {
+      return true;
+    }
+
+    bool areDifferent = false;
+
+    a.forEach((key, value) {
+      if (b[key] != a[key] || (!b.containsKey(key))) {
```
Comment (eusdima) [inline lib/src/core/query_options.dart]: Compare values

Comment (HofmannZ) [reply lib/src/core/query_options.dart]: Nice catch

**Pull request merged**